server/src/main/java/org/elasticsearch/synonyms/SynonymRule.java

```diff
@@ -43,7 +44,7 @@ public class SynonymRule implements Writeable, ToXContentObject {
     private final String id;
 
     public SynonymRule(@Nullable String id, String synonyms) {
-        this.id = id;
+        this.id = Objects.requireNonNullElse(id, UUIDs.base64UUID());
```

Comment:
To make sure an ID is used, I've moved the ID generation method to the actual `SynonymRule` class